### DISCUSSION TITLE
Old UI vs New UI? And Louis Rossmann's opinion about it.

### DISCUSSION THREAD

**Discussion by Bakr-Ali** [blakeblackshear/frigate#14331] [Polls]
Louis Rossmann recently made a [video](https://youtu.be/uiFLqqKkj3M) about how he prefers the old UI before 0.14.0 because of how practical it was and how it showed more information. He will stick with the 0.13.2 version and won't "upgrade". What are your thoughts about the new UI?

**Comment by beatmag** [comment]
I can't even fine tune anything anymore. I have no stats on size or %. So I'm flying blind since 0.14 release. Bounding box is also gone. 

New UI is nice as it becomes more usable but I have no information to work with.

I didn't think it was that hard to have some stats written some where. 

How far is 0.15????? I keep getting birds trigger it.

**Reply by NickM-27** [reply]
0.15 adds this information and more , there should be a beta within the next month or so

**Comment by lawul** [comment]
His points of criticism on the UI seem not that bad and quite fixable to me and are very specific to his exapmle use case "someone got murdered a while ago, and I don't remeber which day of the week it happened".
I don't think the UI was made to enable digging in the past like that, but yeah could be improved. Maybe I don't feel that pain, because I only use Frigate for "what happened today and maybe yesterday". And if it's something noteworthy it's going to be exported.

I agree though, that some of the Buttons and Symbols are not very intuitive (at least in the review area especially on the mobile ui). But you learn where to click after a while.

He also criticitzes the missing scores on events, but those where only showing the score of the events screen shot, not the whole event itselt, or am I wrong? So nothing to base your config fine tuning on(?).
If you feel like your detection config needs to be adjusted, the explroe area (of the current dev build - maybe not in 0.14) is way more helpful for that.

Also, I cannot reproduce the long loading times he experiences. The UI feels very responsive to me.

**Reply by NickM-27** [reply]
Right, the daily review flow is what the new UI mainly addresses right now. The next version of Frigate already has a new Explore page implemented to address this and many other use cases. 

> He also criticitzes the missing scores on events, but those where only showing the score of the events screen shot, not the whole event itselt, or am I wrong? So nothing to base your config fine tuning on(?).

that is incorrect, the score shown in the UI itself was the highest median score of the object which is what is used for threshold filter.

**Comment by blakeblackshear** [comment]
I could have made a similar video with all the complaints for 0.13.2 too. Massive empty portion of the screen and tiny thumbnails in the events list on desktop. You cant "find" much there without clicking to expand each event, and many of them overlap with some lasting for hours and hours. No ability scan through time periods and look for missed objects. Watching a nights worth of footage at 8x speed was painful.

It seems to me like most of the complaints are feature gaps that we knew were being created in the first version of the new UI.

These weren't arbitrary changes dreamed up by a UX designer. We and many others used the dev builds daily for months for the specific use cases we intended to address first. We didn't get it all right. It will continue to improve.

It is helpful to hear from users what they do and don't like about the new UI.

**Reply by OnFreund** [reply]
I believe 0.14 is a huge upgrade vs 0.13 (which I thought was amazing as well). It introduced so many new concepts, that it was clear that some things will be missing initially.
The gaps can be categorized into:
1. Intentional short term omissions. I believe many of these are already addressed in 0.15.
2. Intentional long term omissions.
3. Unintentional long term omissions.

I think a lot of the criticism stems from not understanding which category each person's favorite 0.13 feature falls under, or thinking that a feature in 2 is actually a 3, or should be a 1.
It's clear to me, for example, that "Favorites" is in 2, the reasoning behind it, and the alternative. Not sure it's realistic to have this clarity for every feature.

**Comment by kyle4269** [comment]
At the end of the day, change affects all of us. I’ll admit, I wasn’t a fan of the new UI at first either. But once I took the time to understand its functionality and potential, I’ve grown to really appreciate it. The work Blake, Nick, Josh, and the rest of the contributors have put in is truly impressive. From personal experience, I can say that many of the features I missed from the old UI are being integrated into the new version. We just need to be patient.

**Comment by shinyzard123** [comment]
The new UI is what finally allowed us to drop Blue Iris entirely and move to Frigate. It saves a lot of time and is easier to use. We are looking forward to see what is coming in the next version.

**Reply by UPEngineer** [reply]
I am with you too. The new UI and features let me dump Blue Iris as well.

**Comment by bubba925** [comment]
I prefer the new layout. I didn't at first but I cant even remember how the old layout works anymore. I like being able to review events for the day and export what is important. Mobile use is better as well. Definitely some things that could be improved but overall it serves its purpose and works well.

**Comment by phedders** [comment]
I c/wouldn't go back. Some things I hope/am sure will improve with the new UI but it is *so* much better for daily use. Draggable timeline is a huge win for me and so much faster scanning a days events than before.

**Comment by jvrobert** [comment]
Eh, mixed feelings. I mainly use mobile, and the timeline view is more or less useless, just a squiggly line, looks like a random histogram.  And the layout on mobile is busted, e.g. timeline covers video, you zoom in and then have to find just the right area to zoom back out. Stuff like that.

But I'd say those are implementation details easily fixed, overall I like the approach

**Reply by NickM-27** [reply]
The yellow lines on the timeline show the motion levels that that particular time. Also if the timeline is covering the video then you should report a bug, we've not gotten any reports of that

**Comment by kirk56k** [comment]
I certainly see some major feature gaps with the new UI, but that was absolutely to be expected with a full re-write. Unlike so many UI updates (cough Microsoft cough) that change things just for the sake of change, I see this change as a step in the absolutely correct direction. It is without a doubt an improvement, and I only see it improving more as the feature gaps are filled in. The fact that I can now give my Wife access to this interface and actually expect her to be able to find useful information is a VAST improvement over the previous interface, which I was afraid to show to anyone who wasn't technically inclined. Keep up the good work guys, I very much appreciate it all.

**Comment by drtanzil** [comment]
The new UI is a complete messed up.
If you think its great, you actually don't use frigate day to day use.

It reminds me about windows vista, when Microsoft was claiming how good vista was.

**Reply by bubba925** [reply]
This is an example of completely unconstructive feedback that holds no value.

**Reply by shinyzard123** [reply]
> If you think its great, you actually don't use frigate day to day use.

Lol. You have no idea what you're talking about. If you don't like it, fine, maybe you could make specific suggestions or point out specific issues like many people have here. Suggesting that no one uses it if they like it just makes you appear unreasonable and ignorant.

**Reply by drtanzil** [reply]
@bubba925, @hinyzard123: It might sound unconstructive, unreasonable or ignorant. but that's an honest user review. For constructive, reasonable and informed commentary please go through the video he made. I have same concerns. May be this UI made for high IQ level person like Elon musk or you. it also could be a breakthrough like- Space-X boosted landing, but to me a normal regulars user its a complete messed up of a good app. I'm even unable to revert back to old version (using addon on HA). Feeling pain every day while using new UI.

**Reply by therokh** [reply]
> @bubba925, @hinyzard123: It might sound unconstructive, unreasonable or ignorant. but that's an honest user review. For constructive, reasonable and informed commentary please go through the video he made. I have same concerns. May be this UI made for high IQ level person like Elon musk or you. it also could be a breakthrough like- Space-X boosted landing, but to me a normal regulars user its a complete messed up of a good app. I'm even unable to revert back to old version (using addon on HA). Feeling pain every day while using new UI.

I think the issue people have with your comment, is that you haven't described what is messed up, or not working for you. As a 'normal regulars user' what is it that you want to see Frigate do that it is not doing currently?

**Comment by Anc0dia** [comment]
There are things I like about the redesign - creating/modify masks is 1000% easier/quicker than with 13.2

However, other routine tasks such as reviewing events takes much longer to complete.   I prefer the old event viewer as it is quicker to get the information I needed so I could move on to another task.  Just open the page on my phone, scroll, I could even get everything done using only my thumb.  It was quick.

I find reviewing items in the redesign to be tedious.  Even with version 15 beta, you are list items horizontally, so you then have to select an item to get more information (if it is shown), then reach over and hit back to return to previous page.  It's tedious and slow.  If the item isn't shown then you are selecting options from drop downs etc - it just takes way too long.

I also prefer the smaller thumbnails vs the large snapshots as the large snapshots are pixelated and don't show well on my phone.  I also find that a screen full of large snapshots is too bright at night on the phone.

I'd be happy if you just copied the event page from 13.2 to 15.   It gave all the information quickly that a power user would need.  I don't think the functionality of that page needed to be improved.

**Reply by NickM-27** [reply]
Thanks for the feedback

> I find reviewing items in the redesign to be tedious. Even with version 15 beta, you are list items horizontally, so you then have to select an item to get more information (if it is shown), then reach over and hit back to return to previous page

why are you not just hovering over the thumbnail (or swiping to the right on mobile) to see what happened? In most cases I have no need to click as you can just hover to see what happened. 

Also, in 0.15 you can just click the gray chip if you want more information on that particular review item and all the objects that were detected during that time

> I'd be happy if you just copied the event page from 13.2 to 15. It gave all the information quickly that a power user would need. I don't think the functionality of that page needed to be improved.

Like Blake explained above, there were many things that were inefficient about the events page. It showed each object individually meaning many overlapped or were duplicated leading to more time spent viewing repeat objects. It also required a click every time you wanted to see a snapshot or a clip. This new UI requires no clicks as you can just move your mouse (or swipe to the right on mobile) to see a quick replay of what happened during the event, only requiring a click if you want to view the full res recordings to get a closer look.

**Reply by shinyzard123** [reply]
> This new UI requires no clicks as you can just move your mouse (or swipe to the right on mobile) to see a quick replay of what happened during the event, only requiring a click if you want to view the full res recordings to get a closer look.

This was one of the biggest deals for us as it greatly speeds up the day to day review process. Maybe some users with issues don't know this is possible since the UI doesn't really show you to do that?

**Reply by Anc0dia** [reply]
> This new UI requires no clicks as you can just move your mouse (or swipe to the right on mobile) to see a quick replay of what happened during the event, only requiring a click if you want to view the full res recordings to get a closer look.

Like the person in the video, I'm talking about reviewing items on a phone.   I agree with the reviewer about the old event viewer - all the information was there, zones, % confidence, etc.   And you could quickly scroll through items to find the information you needed.    It was quick and easy.

I've got most of the functionality back of the 13.2 event viewer on 15 by using the Frigate card in HA but I can't submit to plus & it doesn't show % confidence.   But I use it because it's quick for me to review items, and with the smaller thumbnails I can quickly focus on the information in the picture.


![Screenshot_20241015-162312_Home Assistant](https://github.com/user-attachments/assets/502a3fa8-f5f1-4220-a6af-423a63064084)

**Reply by NickM-27** [reply]
> I'm talking about reviewing items on a phone

right, you can swipe from left to right to view the video preview on mobile without needing to tap

> all the information was there, zones, % confidence, etc. And you could quickly scroll through items to find the information you needed. It was quick and easy.

In 0.15 all of that information is visible using the `Review Details` by clicking / tapping on the gray chip

![Screen Shot 2024-10-15 at 14 42 54 PM](https://github.com/user-attachments/assets/5e75c2a0-6320-4f89-bbc6-d60a11333b25)

it's also worth asking to better understand, why do you care specifically what zone the object entered? If you have configured alerts and detections correctly according to your needs then you are guaranteed that an alert is something that happened in a zone you care about and a detection is something that happened in a zone you don't care about. What need is there a need for more specificity at a glance?

**Reply by Anc0dia** [reply]
> it's also worth asking to better understand, why do you care specifically what zone the object entered? If you have configured alerts and detections correctly according to your needs then you are guaranteed that an alert is something that happened in a zone you care about and a detection is something that happened in a zone you don't care about. What need is there a need for more specificity at a glance?

:)  That's a hard one to answer.

Let me put it this way - some people who drive a car would be happy with just a speedometer and a green light on the dashboard that shows everything is running ok.

Other people would like a dashboard that has the tachometer, battery gauge, temperature sensors, gear information etc.  Why?  Personal choice

So in my event viewer I like to be able to see what's happening under the hood in Frigate quickly and efficiently.  And the 13.2 event viewer allowed me to do that.
  
edit: I just want to add that for me, it's just this one item that I miss in the new design. Otherwise I like the redesign and overall the work all the developers have done on Frigate is impressive.

**Reply by NickM-27** [reply]
Thank you for the responses, they are helpful in better understanding different user's use cases, needs, etc.

**Comment by leccelecce** [comment]
I think the 0.14 UI is a vast improvement over 0.13. I will admit though that when I first saw it I wasn't sure.

One point - I am *still* discovering features of the new UI. Just now I found out I could edit the camera group layout grid, because I was re-reading the 0.14 FAQ looking for something else.

The one bit of feedback I'd therefore give is that 0.14 has included a load of new functionality, but it's not always obvious to even daily users. I know the obvious answer here is "read the manual/FAQ", but the reality is that I (like many people) tend to read user guides as a last resort - you just assume you can click around and discover things.

In the example of camera group layout editing, I had never even noticed the three icons in the bottom right of the screen, (despite having viewed it hundreds of times) probably because it's quite an unusual place to put UI controls. Dark mode doesn't help here either. I might suggest moving those controls up to the top bar (which is where most other pages in the UI have them anyway).

![image](https://github.com/user-attachments/assets/0e9224f0-d6cf-4c21-bf24-1358613f5889)

**Comment by kirk56k** [comment]
Having read through a lot of comments and talked to a few people who use this system. I think I might have an idea what the two different, competing, workflows might be. It seems like Workflow #1, is the type of person who has set up Alerts, integrated them with Home Assistant or some other type of notification system, and respond to and review detection events within the hour of their occurrence. This seems to be the workflow that the current UI layout is most designed to support. And for anyone who uses that kind of workflow, the new UI seems to be working out mostly well (aside from the lack of detailed information for the "How the Clock Works" crowd.)

The other workflow, we'll call Workflow #2, seems to revolve around weekend warrior types. These are the people who don't have any type of immediate notification system setup, or don't have the ability or desire to review said notifications, until the weekend (or some much later date). These types of people generally just want to see what's been going on over the last week (or however long it's been since they last reviewed the logs), and the system as it currently stands does make it somewhat difficult to support this type of workflow. I think it would be possible to better support this type of workflow with some minor changes. Perhaps some better options to get quick and easy insight into the last X number of days detection and possibly motion events for a given camera or group of cameras.

Anyway, thats my 1.99 (and 9/10ths of a cent) worth.

**Reply by NickM-27** [reply]
yep, like it says in the original post explaining the new UI, it aimed at the daily review workflow. The review page was never designed to work for the "I know something happened but not sure when" workflow. 

0.15 already has a new Explore page that aims to solve this and many other use cases

**Comment by moandcompany** [comment]
(I'm a fan of this project in general and appreciate everyone's work)

I generally share Louis's point of view with regard to the Frigate 0.14.x major version UI changes. I can also appreciate why a large number of people feel 0.14.x is a huge UI improvement over prior Frigate versions.

- There are many improvements in major version 0.14.x that I appreciate, but the overall style of the day-to-day user interface seems to be designed for someone with different uses cases from my own. It feels difficult to find information or investigate questions of where something might have happened as well as faceted versions of these questions. It seems to be significantly easier to do so with the older interface design, but at the cost of potentially having a lot of detection items to sift through, which is the big tradeoff. 
  - In 0.13.x and prior, conceptually all "detections" are separate "events" even if many of those detections were likely related (e.g. a person riding a bicycle will be detected as two different object types, two different object instances, and accordingly two different events)
  - In 0.14.x+, detections still occur and are recorded separately, due to how the video processing and detection engine works, but are coalesced into conceptual "review items" for the Frigate user interface.
  - The tuning process for camera detection configuration (e.g. motion tuning, zone and other mask definitions) is significantly improved across the board in 0.14.x; there are no regressions in usability here.

- The UI in 0.14.x provides simpler summarizations of detection sets (i.e. "review items") it records, especially for things that have occurred recently on the order of a few hours; this comes at the cost of friction when looking further into the past and across longer units of time.
- Animated video summaries of a particular detection can be useful at a glance, when seen individually. When there are several animations playing simultaneously, this becomes a visual distraction, and this can have negative value. In human factors studies, we've learned that as humans we are wired to take notice of visual changes (e.g. motion); when everything in view has motion, this can be disorienting. In human interface design, this property can positively be used to draw attention to particular areas. (Note: this behavior can be disabled via the "Play Alert Videos" setting for "Live Dashboard" under General Settings)
 - Autoplay of a single video summary during mouse-over is convenient.

I have more than one instance of Frigate for different locations, with both 0.13.x and 0.14.x in some cases. I use these instances in parallel with NVRs and the camera areas covered usually have regular person/vehicle/animal activity. The NVRs are intended to be backup recording sources in case detections were missed or recordings truncated by the Frigate instance.
- In most cases, I prefer to have these systems perform continuous or criteria-driven recording and allow the systems to overwrite old recordings over time as storage runs out. Frigate is used as an index to find specific events or better understand event patterns over time.
- With traditional, commercial-style NVRs and Frigate 0.13.x and older, the concept of "locking" (or starring/flagging) particular recordings exists. The "review item" and "export" concepts in Frigate 0.14.x are intended to accomplish something similar, for persistence of particular recordings, but the implementation seems to create new friction.
 - Put in context, some users want to retain all recordings within a particular time window (e.g. the past thirty days). In this example, they commonly want to lock/flag particular recordings of interest that may be discovered relatively near when those recordings occurred, but also days or weeks after they occurred. This could be for export/download, or for future review. This scenario relates to a workflow where the user has an idea that something may have happened, or has happened, but they do not know exactly when it occurred; they may not have any particular clues related to what they are looking for, e.g. a person wearing a ____, a ____ car, etc. The prompt to investigate what may have happened may occur long after the actual event(s).
  - The investigation process in the above scenario can take multiple iterations, refining criteria of recordings (or detections) to review, such as:  persons during particular hours of the day (e.g. morning, afternoon, evening, late-night) on particular days of the week.


As others point out, the 0.14.x user interface changes appear to have been designed with very different workflows in mind and seems to prioritize users looking for a consumer-style application, prioritizing interaction through a phone/mobile device.
- I imagine that the 0.14.x design changes feel great for users that have a low volume of video recording activity and seldom need to look further back in time than a few days.
- The 0.14.x user interface in the web and mobile web app have almost no text labels to help a user navigate the interfaces or perform actions. I understand the utility of icons and symbology, the principle of how less can be more, and can appreciate that perhaps this helps reduce the need for localization to other languages, but it's also possible to have too little. It feels easy to get lost navigating the 0.14.x interfaces even after repeat use.

**Reply by NickM-27** [reply]
Thanks for the detailed feedback, you may have read in the other replies but 

> they do not know exactly when it occurred; they may not have any particular clues related to what they are looking for, e.g. a person wearing a ____, a ____ car, etc. The prompt to investigate what may have happened may occur long after the actual event(s).
>
> The investigation process in the above scenario can take multiple iterations, refining criteria of recordings (or detections) to review, such as: persons during particular hours of the day (e.g. morning, afternoon, evening, late-night) on particular days of the week.

is one of the main use cases that the Explore view in 0.15 aims to address. Not only with typical date range, camera, label, zone, etc. filters but also (optionally) with smart search powered by CLIP models and genAI if enabled. This makes it very easy to search for something specific, for example (this is just with semantic search enabled, not using genai):

![Screen Shot 2024-10-16 at 15 58 40 PM](https://github.com/user-attachments/assets/e33468b7-453b-4e3c-a481-089850b4254c)

**Reply by moandcompany** [reply]
Thanks Nick,

I started writing my own software to interact with Frigate data back in the 0.12 and earlier days, which relates to some of what was mentioned in this post:  https://github.com/blakeblackshear/frigate/discussions/3676

It's intended to fill in holes that aren't very much aligned with basic NVR functionality, rather to help with understanding event data, surfacing patterns, and finding potential clues.
 
I started exploring adding a semantic search layer, however I believe a nuance of faceted search workflows needs to be considered. CLIP models/etc can be used to generate metadata for "smart" search, however many investigations or explorations start at a much more primitive level. For example, if a vehicle may be involved, there may not be specific knowledge of a particular color, make, or model. Often third-party descriptions of a subject of interest are inaccurate. One of the facets for a "smart" search might be determined over the course of investigation, but the use of such facets occur in the middle or end of an iterative search process. Taking that into mind, surfacing information to improve the start of a search process is extremely valuable (e.g. identifying reoccurrences or patterns, identifying anomalies or deviations from patterns).

Paraphrasing the above:  Improving the process of finding something very specific are valuable, but improving the process of finding/identifying those specifics is also highly valuable.

One part of the problem is finding a specific needle in a haystack, and the other part is understanding what needle is being sought.

**Reply by NickM-27** [reply]
right, I think the filters I listed will help a lot with that initial search or when you don't have a reliable description to start your search off. In any case, the Explore view should help a lot of users who fit the use case where they are looking for something during some unknown period of time and want to iterate searches quickly via filters or some description.

**Comment by Anc0dia** [comment]
The new explore layout setting (unfiltered grid) on edaccd8 looks great on the phone & the snapshot quality seems to have improved.  I prefer this to the 13.2 event viewer.  As always, impressive work!
